linodecli/completion.py

```diff
@@ -15,19 +15,73 @@ def get_completions(ops, help_flag, action):
         return (
             "linode-cli completion [SHELL]\n\n"
             "Prints shell completions for the requested shell to stdout.\n"
-            "Currently, only completions for bash and fish are available."
+            "Currently, only completions for bash, fish, and zsh are available."
         )
     if action == "bash":
         return get_bash_completions(ops)
     if action == "fish":
         return get_fish_completions(ops)
+    if action == "zsh":
+        return get_zsh_completions(ops)
     return (
-        "Completions are only available for bash and fish at this time.\n\n"
+        "Completions are only available for bash, fish, and zsh at this time.\n\n"
         "To retrieve these, please invoke as\n"
-        "`linode-cli completion bash` or `linode-cli completion fish`"
+        "`linode-cli completion bash`, `linode-cli completion fish`, or `linode-cli completion zsh`"
     )
 
 
+def get_zsh_completions(ops):
+    """
+    Generates and returns Zsh shell completions based on the baked spec
+    """
+    completion_template = Template(
+        """#compdef linode-cli linode lin
+
+# This is a generated file by Linode-CLI! Do not modify!
+local -a subcommands
+subcommands=(
+'$subcommands --help'
+)
+
+local -a command
+local -a opts
+
+_arguments -C \\
+  "1: :($subcommands)" \\
+  '*:: :->subcmds' && return 0
+
+if (( CURRENT == 2 )); then
+  case $words[1] in
+    $command_items
+  esac
+fi
+"""
+    )
+
+    command_template = Template(
+        """$command)
+      command=(
+          '$actions --help'
```

Comment:
I'm seeing some weird outputs attempting to autocomplete on subcommands:

```
linode-cli linodes list\ create\ view\ update\ delete\ backups-list\ snapshot\ backups-cancel\ backups-enable\ backup-view\ backup-restore\ boot\ clone\ configs-list\ config-create\ config-view\ config-update\ config-delete\ config-interfaces-list\ config-interface-add\ config-interface-view\ config-interface-update\ config-interface-delete\ config-interfaces-order\ disks-list\ disk-create\ disk-view\ disk-update\ disk-delete\ disk-clone\ disk-reset-password\ disk-resize\ firewalls-list\ ips-list\ ip-add\ ip-view\ ip-update\ ip-delete\ migrate\ upgrade\ nodebalancers\ linode-reset-password\ reboot\ rebuild\ rescue\ resize\ shutdown\ transfer-view\ volumes\ types\ type-view\ --help
```

Dropping the quotes here seems to resolve the issue on my end 👍 

```suggestion
          '$actions --help'
```